source/server/admin/prometheus_stats.cc

```diff
@@ -26,6 +38,20 @@ std::string sanitizeName(const absl::string_view name) {
   return promRegex().replaceAll(name, "_");
 }
 
+/**
+ * Take tag values and sanitize it for text serialization, according to
+ * Prometheus conventions.
+ */
+std::string sanitizeValue(const absl::string_view value) {
+  // Removes problematic characters from Prometheus tag values to prevent
+  // text serialization issues. This matches the prometheus text formatting code:
+  // https://github.com/prometheus/common/blob/88f1636b699ae4fb949d292ffb904c205bf542c9/expfmt/text_create.go#L419-L420.
+  // The goal is to replace '\' with "\\", newline with "\n", and '"' with "\"".
+  std::string tmp = backslashRegex().replaceAll(value, R"(\\\\)");
```

Comment:
I don't think the regexes are buying us much here. What do you think of using [absl::StrReplaceAll](https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_replace.h#L32) to do these substitutions in one pass?

Reply:
I was not aware of StrReplaceAll (somehow missed in a quick skim of docs). Updated.